backend/handlers/math/outcomes/dbpm/marketshares_test.go

```diff
@@ -1,551 +1,709 @@
+			ExpectedScaledPayouts: []int64{20, 20, 0},
+		},
+		{
+			Name: "FourthBetNegativeNoDirection",
 			Bets: []models.Bet{
-				{
-					Amount:   1,
-					Outcome:  "YES",
-					Username: "user2",
-					PlacedAt: now,
-					MarketID: 1,
-				},
-				{
-					Amount:   -1,
-					Outcome:  "YES",
-					Username: "user2",
-					PlacedAt: now.Add(time.Minute),
-					MarketID: 1,
-				},
-				{
-					Amount:   1,
-					Outcome:  "NO",
-					Username: "user1",
-					PlacedAt: now.Add(2 * time.Minute),
-					MarketID: 1,
-				},
-				{
-					Amount:   -1,
-					Outcome:  "NO",
-					Username: "user1",
-					PlacedAt: now.Add(3 * time.Minute),
-					MarketID: 1,
-				},
-				{
-					Amount:   1,
-					Outcome:  "NO",
-					Username: "user1",
-					PlacedAt: now.Add(4 * time.Minute),
-					MarketID: 1,
-				},
-			},
-			ProbabilityChanges: []wpam.ProbabilityChange{
-				{Probability: 0.50},
-				{Probability: 0.75},
-				{Probability: 0.50},
-				{Probability: 0.25},
-				{Probability: 0.50},
-				{Probability: 0.25},
-			},
-			CoursePayouts: []CourseBetPayout{
-				{Payout: 0.25, Outcome: "YES"},
-				{Payout: -0.5, Outcome: "YES"},
-				{Payout: 0.25, Outcome: "NO"},
-				{Payout: -0, Outcome: "NO"}, // golang math.Round() rounds to -0 and +0
-				{Payout: 0.25, Outcome: "NO"},
+				util.GenerateBet(20, "NO", "one", 1, 0),
+				util.GenerateBet(10, "YES", "two", 1, time.Minute),
+				util.GenerateBet(10, "YES", "three", 1, 2*time.Minute),
+				util.GenerateBet(-10, "NO", "one", 1, 3*time.Minute),
 			},
+			CoursePayouts: generateCoursePayouts(
+				[]float64{9.1600000000000001, 2.5, 1.25, 0},
+				[]string{"NO", "YES", "YES", "NO"},
+			),
+			F_YES:                 5.066666666666666,
+			F_NO:                  1.2008733624454149,
+			ExpectedScaledPayouts: []int64{11, 13, 6, 0},
 		},
 	}
 	for _, tc := range testcases {
 		t.Run(tc.Name, func(t *testing.T) {
-			actualPayouts := CalculateCoursePayoutsDBPM(tc.Bets, tc.ProbabilityChanges)
-			if len(actualPayouts) != len(tc.CoursePayouts) {
-				t.Fatalf("expected %d payouts, got %d", len(tc.CoursePayouts), len(actualPayouts))
-			}
+			actualPayouts := CalculateScaledPayoutsDBPM(tc.Bets, tc.CoursePayouts, tc.F_YES, tc.F_NO)
+
+			// Ensure payouts match exactly
 			for i, payout := range actualPayouts {
-				if payout.Payout != tc.CoursePayouts[i].Payout || payout.Outcome != tc.CoursePayouts[i].Outcome {
-					t.Errorf("Test %s failed at index %d: expected %+v, got %+v", tc.Name, i, tc.CoursePayouts[i], payout)
+				if payout != tc.ExpectedScaledPayouts[i] {
+					t.Errorf(
+						"Test %s failed at index %d: expected payout %d, got %d",
+						tc.Name, i, tc.ExpectedScaledPayouts[i], payout,
+					)
 				}
 			}
 		})
 	}
 }
 
-func TestCalculateNormalizationFactorsDBPM(t *testing.T) {
+func TestCalculateExcess(t *testing.T) {
 	testcases := []struct {
-		Name          string
-		F_YES         float64
-		F_NO          float64
-		ExpectedF_YES float64
-		ExpectedF_NO  float64
+		Name           string
```

Comment:
`"Payouts[%d]: got %d, want %d", i, payout, tc.ExpectedScaledPayouts[i]"`
```suggestion
					    "Payouts[%d]: got %d, want %d", i, payout, tc.ExpectedScaledPayouts[i]"
```

Reply:
I don't understand what this is.